controllers/space/space_controller_test.go

```diff
@@ -737,16 +737,126 @@ func TestPromoteSpace(t *testing.T) {
 		})
 	})
 
+	t.Run("tier update (same tier but updated references)", func(t *testing.T) {
+		// given that Space is set to the same tier that has been updated and the corresponding NSTemplateSet is not up-to-date
+		s := spacetest.NewSpace("oddity",
+			spacetest.WithTierNameAndHashLabelFor(basicTier), // assume that at this point, the `TemplateTierHash` label was already removed by the ChangeTierRequestController
+			spacetest.WithSpecTargetCluster("member-1"),
+			spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+			spacetest.WithFinalizer())
+		hostClient := test.NewFakeClient(t, s, basicTier, otherTier)
+		nstmplSet := nstemplatetsettest.NewNSTemplateSet("oddity", nstemplatetsettest.WithReadyCondition()) // NSTemplateSet has references to old basic tier
+		member1Client := test.NewFakeClient(t, nstmplSet)
+		member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+		member2 := NewMemberCluster(t, "member-2", corev1.ConditionTrue)
+		ctrl := newReconciler(hostClient, member1, member2)
+
+		// when
+		res, err := ctrl.Reconcile(context.TODO(), requestFor(s))
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: 1 * time.Second,
+		}, res) // explicitly requeue while the NSTemplate update is triggered by its controller
+		spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
+			Exists().
+			HasTier(basicTier.Name).
+			HasSpecTargetCluster("member-1").
+			HasStatusTargetCluster("member-1").
+			HasConditions(spacetest.Updating()).
+			HasLabel(tierutil.TemplateTierHashLabelKey(basicTier.Name))
```

Comment:
could we also verify the value of the label? at this point, it should be still the older version.
That also means that we need to add a test-case that will verify a situation when:
1. The Space has the hash label value outdated 
2. The NSTemplateSet is up-to-date
3. The controller update the hash label value to the latest one

but there is maybe already such a test-case - not sure :man_shrugging:

Reply:
btw, it would be maybe good to modify
```
HasLabel(tierutil.TemplateTierHashLabelKey(
```
so it automatically checks the value of the label - that it matches the content of the given tier

Reply:
Added a func `HasMatchingTierLabelForTier` that verifies both the label exists and its hash matches: ea242bebc5906482ce2f79135bc2f30ee311d08b

There is already a test named `update not needed` that doesn't set the hash label on the space and the NSTemplateSet is up-to-date which is basically the same thing: https://github.com/codeready-toolchain/host-operator/pull/568/files#diff-92b9cf612f3442ae1d6b7359f59a2c5683ab7edb129c96fdc050bd4c083e5671R830